web/client-api/src/main/java/io/deephaven/web/client/api/widget/plot/JsFigure.java

```diff
@@ -638,37 +638,31 @@ public Promise fetch(JsFigure figure, FetchObjectResponse response) {
 
             Promise<?>[] promises = new Promise[response.getTypedExportIdList().length];
 
-            response.getTypedExportIdList().forEach((p0, p1, p2) -> {
-                if (!p0.getType().equals("Table")) {
+            response.getTypedExportIdList().forEach((ticket, index, array) -> {
+                if (!ticket.getType().equals("Table")) {
                     // TODO (deephaven-core#62) implement fetch for tablemaps
-                    assert false : p0.getType() + " found in figure, not yet supported";
+                    assert false : ticket.getType() + " found in figure, not yet supported";
                     return null;
                 }
 
 
                 // Note that creating a CTS like this means we can't actually refetch it, but that's okay, we can't
                 // reconnect in this way without refetching the entire figure anyway.
-                promises[p1] = Callbacks.<ExportedTableCreationResponse, Object>grpcUnaryPromise(c -> {
-                    connection.tableServiceClient().getExportedTableCreationResponse(p0.getTicket(),
+                promises[index] = Callbacks.<ExportedTableCreationResponse, Object>grpcUnaryPromise(c -> {
+                    connection.tableServiceClient().getExportedTableCreationResponse(ticket.getTicket(),
                             connection.metadata(),
                             c::apply);
                 }).then(etcr -> {
                     ClientTableState cts = connection.newStateFromUnsolicitedTable(etcr, "table for figure");
                     JsTable table = new JsTable(connection, cts);
                     // never attempt a reconnect, since we might have a different figure schema entirely
                     table.addEventListener(JsTable.EVENT_DISCONNECT, ignore -> table.close());
+                    tables[index] = table;
```

Comment:
This was previously `tables[tables.length] = table` when it should have been using `p1` for the subscript as line 651 does. I changed the names of the lambda params to make this more obvious.